app/src/main/java/de/westnordost/streetcomplete/screens/about/LogsFragment.kt

```diff
@@ -97,7 +62,7 @@ class LogsFragment : TwoPaneDetailFragment(R.layout.fragment_logs) {
     }
 
     private fun onClickShare() = viewLifecycleScope.launch {
-        val logText = getLogs().format()
+        val logText = viewModel.logs.value.format()
         val logTimestamp = LocalDateTime.now().toString()
         val logTitle = "${BuildConfig.APPLICATION_ID}_${BuildConfig.VERSION_NAME}_$logTimestamp.log"
```

Comment:
Replacement with a ViewModel function of `getLogText` & `getLogTitle` would make this cleaner for the UI.

This is business code anyways (producing a state)

Reply:
I see your point, but I'd like to keep view models platform agnostic and this method to share the logs is specific to Android.

I'm not sure what to do yet with platform specific code in the `Fragment`s, as they certainly have no place in Compose code either.
I think I am going to postpone thinking about this when the time actually comes. You're probably right that in the end, it will end up one way or another in the ViewModel, with some `expect` / `actual`.

Now that you made me think about it... for starting an Activity (to send an Intent), you _need_ a `context` but at the same time, docs say that the activity etc. should not be a member in a ViewModel, so I do wonder how it would have to look in the end.